src/lib.rs

```diff
@@ -70,8 +70,21 @@ pub mod log;
 pub mod tracing;
 
 /// Logging flags to `#[command(flatten)]` into your CLI
-#[derive(clap::Args, Debug, Clone, Default)]
+#[derive(clap::Args, Debug, Clone, Default, PartialEq, Eq)]
```

Comment:
This PR seems to be mixing a lot of different things
- Making types support `Eq`
- More `From` impls
- A refactor with how the count math is done

etc

Ideally these would at least be split into individual commits
- We can track them in release notes more easily
- I can better see how all of the parts for a single change fit together